### PR TITLE
Merge documentation of apps_mem module

### PR DESCRIPTION
**Pull request by quic-bkumar**
1) Adding new documents for the apps_mem module
2) Fix msg packet details on adspmsgd

### CODE REVIEW

File: Docs/adspmsgd.md [hunk 1]
```diff
@@ -29,19 +29,16 @@ typedef struct {
 ## 4. Msg/Packet format
 In ADSPMSGD, the shared buffer contains multiple string messages from DSP. Each message is stored in the buffer and is ended with a null character to indicate its conclusion. This format allows the system to efficiently store, read, and manage multiple messages within a single shared buffer.
```
Comment (ekanshibu) [inline Docs/adspmsgd.md:31]: I think you have made some changes to adspmsgd docs also, can this go in as a separate patch?

**Comment by quic-bkumar** [comment]
Dropping off, as a latest pull request is created.
